WordPress/gutenberg
# Search block: Add typography supports

Related:
- https://github.com/WordPress/gutenberg/issues/43241
- https://github.com/WordPress/gutenberg/issues/43242

## What?

Adds typography supports to the Search block.

## Why?

- Allows typographic styling of the Search block.
- Improves consistency of our design tools across blocks.

## How?

- Opts into typography supports and applies them to the: 
    - label
    - input (except text decoration)
    - button
- Sets the font size control to display by default matching most other blocks.

## Testing Instructions

1. Edit a post, add a Search block, and select it. (See example HTML below).
2. Check that all typography controls are now available.
3. Test various typography settings ensuring styles are applied in the editor.
4. Test various search block manifestations, e.g., button position, icon button.
5. Save and confirm that the frontend appears as expected
6. Do the same in the site editor. Also, edit global styles and make sure the styles are applied.
7. Style the Search block via theme.json (see example json below).
8. Confirm the theme.json style applies correctly in the editor and frontend.

<details>

<summary>Example block code</summary>

```html



<hr class="wp-block-separator has-alpha-channel-opacity is-style-wide"/>





<hr class="wp-block-separator has-alpha-channel-opacity is-style-wide"/>





<hr class="wp-block-separator has-alpha-channel-opacity is-style-wide"/>





<hr class="wp-block-separator has-alpha-channel-opacity is-style-wide"/>




```

</details>

Example theme.json snippet.

```json
	"styles": {
		"blocks": {
			"core/search": {
				"typography": {
					"textDecoration": "line-through"
					"letterSpacing": "12px",
					"fontSize": "2rem",
					"textTransform": "uppercase",
					"lineHeight": "1",
					"fontStyle": "italic",
					"fontWeight": "600"
				}
			}
		}
	},
```

## Screenshots or screencast 


https://user-images.githubusercontent.com/6458278/186588035-43e0c91c-10c8-4bd5-90bf-5937c508a878.mp4
